setup.sh

```diff
@@ -69,6 +69,11 @@ check_dependencies() {
         exit 1
     fi
 
+    # For convenience, let's make sure github is in the known-hosts
+    # file as well.  This is a noop if it's already in there.
+    mkdir -p ~/.ssh
+    ssh-keyscan -H github.com
```

Comment:
Does this add the key to `.ssh/known_hosts`? I was under the impression that it just printed the keys out. I agree that whatever the solution is, it would be a no-op if the keys are already present.

(TBH, the original code with an inline, hard coded key seems fine to me.)

Reply:
No, it adds them to known_hosts.  We are using this formulation already in all our dockerfiles.

Reply:
Hmm. I just checked with the following:
```
cd .ssh
mv known_hosts known_hosts.orig
ssh-keyscan -H github.com
```
And no known_hosts was created.
I also tried creating an empty file, just in case:
```
touch known_hosts
chmod 600 known_hosts
ssh-keyscan -H github.com
cat known_hosts
```
And still nothing.

Reply:
Doh, you are so right!  I mis-copied.  It should be `ssh-keyscan github.com >> ~/.ssh/known_hosts`

Reply:
Ah, okay. With the append, what do you think about keeping `grep -q github.com ~/.ssh/known_hosts 2>/dev/null ||` from above so they keys aren't added multiple times when running the dot files?